ON-1240 ILR data fields for data accuracy

arshedmust opened the pull request.

das-build-agent committed:
Added new fields to SubmissionEvent FundModelDelLocPostCodeLearnActEndDateWithdrawReasonOutcome AimType

carlbm commented on src/SharedPipelineComponents/IlrSubmissionEvents/SFA.DAS.Provider.Events.Submission/SubmissionEventsProcessor.cs:
```diff
@@ -109,16 +109,6 @@ private SubmissionEvent CalculateDelta(IlrDetails currentIlr, IlrDetails lastSee
                 (@event = @event ?? new SubmissionEvent()).PathwayCode = currentIlr.PathwayCode;
             }
 
-            if (currentIlr.ActualStartDate != lastSeenIlr?.ActualStartDate)
```
Why was this removed?

alandburns replied:
I believe it is because they are being handled at line 139-140 instead,
                || currentIlr.ActualStartDate != lastSeenIlr?.ActualStartDate
                || currentIlr.PlannedEndDate != lastSeenIlr?.PlannedEndDate
but if this is likely to be an issue elsewhere the original lines 112 - 121 can remain.

alandburns replied:
As there is no impact, this will be reverted

carlbm reviewed (changes requested):
Curious about the removed logic

arshedmust committed:
Update SubmissionEventsProcessor.cs

Reverted
 if (currentIlr.ActualStartDate != lastSeenIlr?.ActualStartDate)
            {
                (@event = @event ?? new SubmissionEvent()).ActualStartDate = currentIlr.ActualStartDate;
            }

            if (currentIlr.PlannedEndDate != lastSeenIlr?.PlannedEndDate)
            {
                (@event = @event ?? new SubmissionEvent()).PlannedEndDate = currentIlr.PlannedEndDate;
            }

alandburns commented:
Closing this branch as we will not be implementing the changes to Submission Events and Provider Events with the closing of Payments V1. And Payments V2 will not be supporting the data we need.
